### PR TITLE
TG-5184: INTERNAL REVIEW: Proposed tests for Mojang-DataFixerUpper

### PR DESCRIPTION
**Pull request by JohnLBergqvist**

### CODE REVIEW

File: src/test/java/com/mojang/datafixers/DataFixUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    final int retval = DataFixUtils.makeKey(version, subVersion);
+    // Assert result
+    Assert.assertEquals(0, retval);
+  }
+
+  /*
+  * Test generated by Diffblue Deeptest.
+  * This test case covers the entire method.
+  */
+  @Test
```
Comment (sudo-jaa) [inline src/test/java/com/mojang/datafixers/DataFixUtilsTest.java:57]: Is there another test for smallestEncompassingPowerOfTwo that has a non-0 output?

File: src/test/java/com/mojang/datafixers/DataFixUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  /* testedClasses: com/mojang/datafixers/DataFixUtils.java */
+
+  /*
+  * Test generated by Diffblue Deeptest.
+  * This test case covers:
+  *  - conditional line 36 branch to line 36
+  */
+  @Test
```
Comment (sudo-jaa) [inline src/test/java/com/mojang/datafixers/DataFixUtilsTest.java:16]: :+1:

File: src/test/java/com/mojang/datafixers/DataFixUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    Assert.assertEquals(8, retval);
+  }
+
+  /*
+  * Test generated by Diffblue Deeptest.
+  * This test case covers:
+  *  - conditional line 36 branch to line 36
+  */
+  @Test
+  public void ceillog2InputPositiveOutputPositive2() {
```
Comment (sudo-jaa) [inline src/test/java/com/mojang/datafixers/DataFixUtilsTest.java:27]: :+1:

File: src/test/java/com/mojang/datafixers/DataFixUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    // Assert result
+    Assert.assertEquals(8, retval);
+  }
+
+  /*
+  * Test generated by Diffblue Deeptest.
+  * This test case covers the entire method.
+  */
+  @Test
+  public void makeKeyInputZeroOutputZero() {
```
Comment (sudo-jaa) [inline src/test/java/com/mojang/datafixers/DataFixUtilsTest.java:37]: Is there a test for makeKey that demonstrates a non-0 output?

File: src/test/java/com/mojang/datafixers/DataFixUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    // Assert result
+    Assert.assertEquals(0, retval);
+  }
+
+  /*
+  * Test generated by Diffblue Deeptest.
+  * This test case covers the entire method.
+  */
+  @Test
+  public void makeKeyInputZeroZeroOutputZero() {
```
Comment (sudo-jaa) [inline src/test/java/com/mojang/datafixers/DataFixUtilsTest.java:47]: Is there a test for makeKey that demonstrates a non-0 output?

File: src/test/java/com/mojang/datafixers/DataFixUtilsTest.java (hunk 1 above)
Comment (JohnLBergqvist) [reply src/test/java/com/mojang/datafixers/DataFixUtilsTest.java:57]: No. Please infer that if there is only 1 test there when you think there should be more, that that was the only valid test that was generated.

File: src/test/java/com/mojang/datafixers/DataFixUtilsTest.java (hunk 4 above)
Comment (JohnLBergqvist) [reply src/test/java/com/mojang/datafixers/DataFixUtilsTest.java:37]: No. Please infer that if there is only 1 test there when you think there should be more, that that was the only valid test that was generated.

File: src/test/java/com/mojang/datafixers/DataFixUtilsTest.java (hunk 5 above)
Comment (JohnLBergqvist) [reply src/test/java/com/mojang/datafixers/DataFixUtilsTest.java:47]: No. Please infer that if there is only 1 test there when you think there should be more, that that was the only valid test that was generated.

**Comment by paulkennethkent** [comment]
> There are at least some classes where the methods are public but the constructor is private, resulting in reflection being used (presumably validly). If you'd like me to include them as well, I can, but it will take me longer as i'd have to sift through the classes by hand to work out if any reflection used is genuinely necessary.

If these tests are accepted then we can try with the reflection tests next. 

👍 they all look good.

**Review by paulkennethkent — APPROVED** [review]

**Comment by petersear** [comment]
Needs adding to spreadsheet and MIT licence included(?)

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**